app/helpers/budget_bar_helper.rb

```diff
@@ -5,15 +5,30 @@ def progress_bar_color(type)
     "progress-bar-#{color}"
   end
 
-  def progress_bar_width(value)
-    "width: #{to_pct(value)}%"
+  def progress_bar_width(value, total = 1_000)
+    "width: #{to_pct(value, total)}%"
+  end
+
+  def budget_bar_expenditure_content(patient_hash)
+    link = link_to patient_hash[:name], edit_patient_path(patient_hash[:id])
+    appt_text = patient_hash[:appointment_date] ?
+      "appt on #{patient_hash[:appointment_date]&.display_date}" :
+      'no appt date'
+    safe_join([link, appt_text], ' - ')
+  end
+
+  def budget_bar_remaining(expenditures, limit)
```

Comment:
This is some dense code in this here function! I feel like I'm back in Perl!

Some clarification comments on the logic behind this would be welcome, as I don't know how to check this does what is intended.

Reply:
haha, sorry, definitely my fault. I've added a few comments that I hope are helpful and adjusted the filter a little bit.